### ISSUE TITLE
[Build] Fails on arm64: error: no member named 'linux_id' in 'cpuinfo_processor'

### ISSUE THREAD

**Issue by yurivict** [microsoft/onnxruntime#23181]
### Describe the issue

Build fails.

### Target platform

FreeBSD 14.1

### Build script

cmake && gmake

### Error / output

```
/wrkdirs/usr/ports/misc/onnxruntime/work/onnxruntime-1.18.2/onnxruntime/core/platform/posix/env.cc:293:38: error: no member named 'linux_id' in 'cpuinfo_processor'
  293 |           th_aff.push_back(log_proc->linux_id);
      |                            ~~~~~~~~  ^
1 error generated.
```

[log](https://pkg-status.freebsd.org/ampere2/data/main-arm64-default/pa22ab6fdaffb_sb762b199afc/logs/onnxruntime-1.18.2_1.log)

### GCC / Compiler Version

clang-18

**Comment by snnn** [comment]
Yeah, it is Linux only.  I think we need to change this line  https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/platform/posix/env.cc#L43 to use an explicit allowlist

```
#if defined(CPUINFO_SUPPORTED) && defined(__linux__) && !defined(__ANDROID__) 
#include <cpuinfo.h>
#define ORT_USE_CPUINFO
#endif
```

What do you think?

Actually, __ANDROID__ might work.  We should try to enable it. @skottmckay , FYI.

**Comment by skottmckay** [comment]
Why is `__linux__` not defined if you're building on FreeBSD? 

https://github.com/pytorch/cpuinfo/blob/ca156f7bc9109c552973414a63d310f76ef0cbf8/include/cpuinfo.h#L615-L623

We have less restrictions in the #include of cpuinfo here.

https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/common/cpuid_info.cc#L53-L54

And the github readme from https://github.com/pytorch/cpuinfo says 
>Cross-platform availability:
Linux, Windows, macOS, Android, iOS and FreeBSD operating systems
x86, x86-64, ARM, and ARM64 architectures

Given that, I would expect this works for at least ANDROID and hopefully macOS/iOS as well.

**Comment by yurivict** [comment]
> Why is __linux__ not defined if you're building on FreeBSD?

FreeBSD has __FreeBSD__ defined, not __linux__.
The problem only exists on arm64, it builds fine on other architectures.

**Comment by yurivict** [comment]
This isn't a stale issue.
Builds are still breaking due to this issue.

**Comment by snnn** [comment]
But it needs to be fixed in Pytorch's CPUINFO repo.  Or you may build ONNX Runtime without this library. Or, you may maintain a local patch for this issue in FreeBSD ports?
